### ISSUE TITLE
PouchDb `Failed to execute 'transaction' on 'IDBDatabase'` due to device inactivity

### ISSUE THREAD

**Issue by dianabarsan** [medic/medic-webapp#8149]
**Describe the bug**
This issue was reported numerous times, but became prominent more recently, after a big project upgraded to a newer version of the CHT, and started writing tasks to disk. 
Upon inspecting feedback documents generated by users reporting this problem, we've noticed that when a "global failure" was reported, the log list frequently also contained errors about IndexedDb timeouts.  
An example: 
```
\"Error getting tasks for all contacts\",{\"status\":500,\"name\":\"indexed_db_went_bad\",\"message\":\"unknown\",\"error\":true,\"reason\":\"TimeoutError\"}]"}
```

We followed this lead and found several Chromium bug reports about people seeing similar errors when laptops went to sleep while IndexedDB connections were open: 
https://bugs.chromium.org/p/chromium/issues/detail?id=855624
https://bugs.chromium.org/p/chromium/issues/detail?id=955470

@kennsippell 's excellent data analysis showed that:
- this is happening when users have a large number of documents
- issue is more frequent for older versions of Chrome but there are cases for newer versions as well. 

I managed to replicate this multiple times on my own device (Pixel 6a), on CHT-Core ~4.2.0, with a user with 26000 documents. 

**To Reproduce**
1. Replicate a user with a large number of documents (according to data, above 25000 docs)
2. Open the task page.
3.1. Put the phone to sleep for a few minutes
3.2. Switch to another activity for a few minutes (I had a whatsapp call which probably took a lot of device resources). 
4. Open CHT App again and see error. 

**Expected behavior**
Tasks should be displayed. 

**Logs**
![image](https://user-images.githubusercontent.com/35681649/227868362-d7722875-bec7-4cdd-81a5-cf85eb23f3d9.png)

The last error seems to repeat ad-infinitum. I kept the tab opened for 30 minutes, and the console reports that 36.000 have been logged: 

![image](https://user-images.githubusercontent.com/35681649/227868610-a07a0684-869e-4b44-83a1-a575117d8411.png)

This is a sample feedback document that was generated during one of these events: 

https://gist.github.com/dianabarsan/82c01f0c3331f7c85427151816c9cc3c

**Screenshots**
![image](https://user-images.githubusercontent.com/35681649/227868161-65bf8401-ba3f-4e29-959f-949c9d00cfaa.png)

If I ignore the error and stay on the tasks page I see: 
![image](https://user-images.githubusercontent.com/35681649/227966337-6f44996f-3990-4d10-b6ec-a15dae60a60a.png)

**Environment**
- Browser: Chrome
- App: webapp
- Version: TBD based on data analysis

**Additional context**
This is definitely a race condition, so repro steps will most likely not trigger the error every time. 
Refreshing the app and keeping the screen on and the app in the foreground fixes the issue.

**Comment by dianabarsan** [comment]
I managed to trigger another error, using the same method: 

![image](https://user-images.githubusercontent.com/35681649/227939624-487ca0ca-eda9-43a2-8baa-5e23b2eb573f.png)

**Comment by dianabarsan** [comment]
![image](https://user-images.githubusercontent.com/35681649/227942436-28bad01e-219d-4b6d-8e0f-14ee16afda76.png)

**Comment by garethbowen** [comment]
I can't help but think if when we catch that error we just invalidate the [db cache](https://github.com/medic/cht-core/blob/master/webapp/src/ts/services/db.service.ts#L20) and re-register the [db.watch in the changes service](https://github.com/medic/cht-core/blob/master/webapp/src/ts/services/changes.service.ts#L75), then the app may just recover, possibly without the user noticing anything at all.

**Comment by dianabarsan** [comment]
> re-register, possibly without the user noticing anything at all.

My worry about this approach is that the device will end up in an infinite loop of timing out and retrying, without the user noticing anything at all. I think we should do the opposite, and make it more obvious about what should be done: keep the phone screen on and the app in the foreground.

**Comment by garethbowen** [comment]
> I think we should do the opposite

Fair point. Maybe it comes down to how common it is per user? If we see it happening all the time for one user then we should give them instructions to fix it, but if it's random and rare for any given user then I'd go for graceful recovery.

**Comment by kennsippell** [comment]
Some historic data on this issue at https://github.com/medic/cht-core/issues/8155#issuecomment-1490679812
